### PR TITLE
chore: StepType & AssertedEventType enums

### PR DESCRIPTION
**Pull request by almog-geva**
Added new StepType & AssertedEventType enums to replace all hard-coded strings throughout the code and restrict possible values to StepType/AssertedEventType (instead of a 'string') wherever needed

- [x] Tests pass

### CODE REVIEW

**Comment by puskuruk** [comment]
Do we really need that? Because typescript already handles which strings you can use with unions with its current implementation.

**Comment by almog-geva** [comment]
@puskuruk really need? no, it's a chore. But-

1. Does it really make more sense to hard-code values throughout the entire code?
2. Enums are useful when setting properties or values that can only be a certain number of possible values. I believe this is exactly the case. As of now, 'type' is just a string (as in the BaseStep interface), which I think is incorrect.
3. In my personal project, it was necessary for me to maintain a list of all possible values both for validation and autocomplete. With an exported enum from this library, I wouldn't have to worry about future changes.

File: test/stringify.test.ts [hunk 1]
```diff
@@ -156,14 +156,14 @@ describe('stringify', () => {
       title: 'Test Recording',
       steps: [
         {
-          type: 'scroll' as const,
+          type: StepType.Scroll as const,
```
Comment (OrKoN) [inline test/stringify.test.ts:159]: would we still need `as const` here and in other places?

**Review by OrKoN — APPROVED** [review]

File: test/stringify.test.ts (hunk 1 above)
Comment (almog-geva) [reply test/stringify.test.ts:159]: It was surprising to me as well, but yes, still needed.

image

**Pull request merged**